emu/memory.c

```diff
@@ -256,10 +256,13 @@ void *mem_ptr(struct mem *mem, addr_t addr, int type) {
         entry = mem_pt(mem, page);
     }
 
-    if (entry != NULL && type == MEM_WRITE) {
+    if (entry != NULL && (type == MEM_WRITE || type == MEM_WRITE_PTRACE)) {
         // if page is unwritable, well tough luck
-        if (!(entry->flags & P_WRITE))
+        if (type != MEM_WRITE_PTRACE && !(entry->flags & P_WRITE))
             return NULL;
+        if (type == MEM_WRITE_PTRACE) {
+            entry->flags |= P_WRITE | P_COW;
```

Comment:
```suggestion
            // TODO: Is P_WRITE really correct? The page shouldn't be writable without ptrace.
            entry->flags |= P_WRITE | P_COW;
```

Reply:
I'm unsure what the concern is here?

Reply:
I think if P_WRITE is set that will transmute a read-only page into a page that can be written by userspace, which doesn't seem right. Doesn't need to be fixed right away, hence the suggestion of a TODO.